website/oauth/models/__init__.py

```diff
@@ -443,3 +443,63 @@ def absolute_api_v2_url(self):
     # used by django and DRF
     def get_absolute_url(self):
         return self.absolute_api_v2_url
+
+class ApiOAuth2PersonalToken(StoredObject):
+    """
+    Store information about recognized OAuth2 scopes. Only scopes registered under this database model can
+        be requested by third parties.
+    """
+    _id = fields.StringField(primary=True,
+                             default=lambda: str(ObjectId()))
+
+    token_id = fields.StringField(default=lambda: uuid.uuid4().hex,  # Not *guaranteed* unique, but very unlikely
+                               unique=True)
+
+    owner = fields.ForeignField('User',
+                                backref='created',
+                                index=True,
+                                required=True)
+
+    user_id = fields.StringField(default=owner)
+
+    name = fields.StringField(required=True, index=True)
+
+    scopes = fields.StringField(required=True)
+
+    date_last_used = fields.DateTimeField(auto_now_add=datetime.datetime.utcnow,
```

Comment:
After discussion with Michael, this data is tracked in CAS but not exposed to the OSF. Submitted low priority ticket to make this available, removing this from UI for now.